tests/unit-tests/rest-api/class-sensei-rest-api-messages-controller.php

```diff
@@ -0,0 +1,243 @@
+<?php
+
+/**
+ * Class Sensei_REST_API_Messages_Controller tests.
+ */
+class Sensei_REST_API_Messages_Controller_Tests extends WP_Test_REST_TestCase {
+
+	/**
+	 * A server instance that we use in tests to dispatch requests.
+	 *
+	 * @var WP_REST_Server $server
+	 */
+	protected $server;
+
+	/**
+	 * Test specific setup.
+	 */
+	public function setUp() {
+		parent::setUp();
+
+		global $wp_rest_server;
+		$wp_rest_server = new WP_REST_Server();
+		$this->server   = $wp_rest_server;
+
+		do_action( 'rest_api_init' );
+
+		// We need to re-instansiate the controller on each tests to register any hooks.
+		new Sensei_REST_API_Messages_Controller( 'sensei_message' );
+	}
+
+	/**
+	 * Test specific teardown.
+	 */
+	public function tearDown() {
+		parent::tearDown();
+
+		global $wp_rest_server;
+		$wp_rest_server = null;
+	}
+
+	/**
+	 * Class wide setup.
+	 *
+	 * @param WP_UnitTest_Factory $factory Helper factory to create WP objects.
+	 */
+	public static function wpSetUpBeforeClass( $factory ) {
+		$teacher_role = new Sensei_Teacher();
+		$teacher_role->create_role();
+	}
+
+
+	/**
+	 * Tests if privileged users can access all messages.
+	 *
+	 * @since 2.3.0
+	 * @covers Sensei_REST_API_Messages_Controller::exclude_others_comments
+	 */
+	public function testPrivilegedUsersCanAccessMessages() {
+
+		$this->factory->user->create(
+			array(
+				'role'       => 'subscriber',
+				'user_login' => 'login',
+			)
+		);
+
+		$this->create_sensei_message( 'Message title', 'login' );
+
+		$teacher_id = $this->factory->user->create( array( 'role' => 'teacher' ) );
+		wp_set_current_user( $teacher_id );
+
+		$request  = new WP_REST_Request( 'GET', '/wp/v2/messages' );
+		$response = $this->server->dispatch( $request );
+
+		$this->assertCount( 1, $response->get_data() );
+		$this->assertEquals( 'Message title', $response->get_data()[0]['displayed_title'] );
+
+		$admin_id = $this->factory->user->create( array( 'role' => 'administrator' ) );
+		wp_set_current_user( $admin_id );
+
+		$request  = new WP_REST_Request( 'GET', '/wp/v2/messages' );
+		$response = $this->server->dispatch( $request );
+
+		$this->assertCount( 1, $response->get_data() );
+		$this->assertEquals( 'Message title', $response->get_data()[0]['displayed_title'] );
+	}
```

Comment:
For tests like this one that have several assertions, it is conventional to add a message to the call. This makes it easier to see what failed when a test failure occurs.

For example: https://github.com/Automattic/sensei-wc-paid-courses/blob/master/tests/unit-tests/includes/test-class.courses.php#L237

Looks like several of our tests actually don't do this, but I think it's still a good practice.